Library/Formula/libtbox.rb

```diff
@@ -0,0 +1,90 @@
+class Libtbox < Formula
+  desc "The Cross-platform C development library"
+  homepage "https://github.com/waruqi/tbox"
+  url "https://github.com/waruqi/tbox/archive/v1.5.1.tar.gz"
+  mirror "http://tboox.net/release/tbox/tbox-v1.5.1.tar.gz"
+  sha256 "9a960a7b362c0a7f694aee505a1ab4d6677cd54542bcf200eda4fcb56f97f76b"
+  head "https://github.com/waruqi/tbox.git"
+
+  option "enable-debug", "Build with debug support"
+
+  option "without-zlib", "Disable Zlib compression library"
+  option "without-pcre", "Disable Pcre regular expressions library"
+  option "without-pcre2", "Disable Pcre2 regular expressions library with a new API"
+  option "without-mysql", "Disable Mysql database"
+  option "without-sqlite", "Disable Sqlite database"
+  option "without-openssl", "Disable Openssl SSL/TLS cryptography library"
+  option "without-mbedtls", "Disable Mbedtls Cryptographic & SSL/TLS library"
+
+  option "without-xml", "Disable Xml module"
+  option "without-zip", "Disable Zip module"
+  option "without-asio", "Disable Asio module"
+  option "without-regex", "Disable Regular expressions module"
+  option "without-object", "Disable Object module"
+  option "without-thread", "Disable Thread module"
+  option "without-network", "Disable Network module"
+  option "without-charset", "Disable Charset module"
+  option "without-database", "Disable Database module"
+
+  depends_on "pcre" => :optional
+  depends_on "pcre2" => :optional
+  depends_on "lzlib" => :optional
+  depends_on "mysql" => :optional
+  depends_on "sqlite" => :optional
+  depends_on "openssl" => :optional
+  depends_on "mbedtls" => :optional
```

Comment:
Using `:optional` means that it won’t be installed unless one uses `--with-foo`. This means that e.g. the `--without-pcre` option you created above won’t have any effect because by default `pcre` won’t be installed. The opposite is `:recommended`.